src/app/client/src/app/modules/dial-code-search/services/dial-code/dial-code.service.ts

```diff
@@ -33,6 +33,20 @@ export class DialCodeService {
         map(apiResponse => _.get(apiResponse, 'result')));
   }
 
+  /**
+   * makes API call to search for dialCode
+   */
+  public searchDialCodeAssemble(dialCode: string, online: boolean): Observable<any[]> {
```

Comment:
rename this function to searchDialCode and call this.publicDataService.post(option), directly instead of calling  searchService. contentDialCodeAssembleSearch contentDialCodeAssembleSearch is not needed